py/src/braintrust/logger.py

```diff
@@ -197,14 +201,33 @@ def flush(self, initial_items=None, batch_size=100):
                     break
 
             if len(items) > 0:
-                conn.post_json("logs", items)
+                # Construct batches of items that do not exceed the max request size divided by 2
+                curr = []
+                curr_len = 0
+                for item in items:
+                    item_s = json.dumps(item)
+                    item_len = len(item_s)
+                    if curr_len + item_len > MAX_REQUEST_SIZE / 2 and len(curr) > 0:
+                        response_raise_for_status(conn.post("/logs", data=construct_json_array(curr)))
+                        curr = []
+                        curr_len = 0
+
+                    curr.append(item_s)
+                    curr_len += item_len
+
+                if len(curr) > 0:
+                    response_raise_for_status(conn.post("/logs", data=construct_json_array(curr)))
 
             if len(items) < batch_size:
                 break
 
             items.clear()
 
 
+def construct_json_array(items):
```

Comment:
nit: maybe for uniformity define this right under `MAX_REQUEST_SIZE` as is done in js?